### PR TITLE
Bronwyn frequency encoder and basic test implementation

### LINKED ISSUE #25: Code frequency_encoder
This is just an issue to track the progress of the entire function (since the milestone instructions state to create an issue for each function, and we only created issues for each docstring in particular).

- [x] Docstring
- [x]  Code

Will modify as we understand what we need to do in the coming weeks.

### PR DESCRIPTION
**Pull request by bbaillie**

### CODE REVIEW

File: R/frequency_encoder.R [hunk 1]
```diff
@@ -1,13 +1,18 @@
+
+encode_freq <- function(X) {
+  X
+}
+
 #' frequency_encoder
 #'
 #' This function encodes categorical variables using the frequencies of each category.
 #'
```
Comment (robilizando) [inline R/frequency_encoder.R:22]: Maybe add a quick 3 x 3 example of what the frequency encoder does so users can relate to what happen to their data frame?

image

Comment (robilizando) [reply R/frequency_encoder.R:22]: I am speculating that this is what the function does.

File: R/frequency_encoder.R [hunk 2]
```diff
@@ -1,13 +1,18 @@
+
+encode_freq <- function(X) {
+  X
+}
```
Comment (robilizando) [inline R/frequency_encoder.R]: So what is X here, the data?.

Never mind, here is where you do the magic :)

So I guess we need to put this function below the docstring don't we?

Comment (robilizando) [reply R/frequency_encoder.R]: Or maybe inside the frequency encoder function so we do not have to write two functions?..

File: R/frequency_encoder.R [hunk 3]
```diff
@@ -16,7 +21,8 @@
 #' my_test,
 #' cat_columns = c("foo"))
 frequency_encoder <- function(X_train, X_test, cat_columns) {
-
```
Comment (robilizando) [inline R/frequency_encoder.R:19]: Do not forget to write a defensive test, like check that X_train and X_test are tibbles or data.frames and cat_columns is a list.

File: R/frequency_encoder.R [hunk 4]
```diff
@@ -21,8 +23,19 @@ encode_freq <- function(X) {
 #' my_test,
 #' cat_columns = c("foo"))
 frequency_encoder <- function(X_train, X_test, cat_columns) {
-  train_processed <- dplyr::mutate_if(X_train,~is.element(deparse(substitute(.x)), cat_columns), encode_freq)
```
Comment (robilizando) [inline R/frequency_encoder.R]: I realized I commented in a previous edit of the function. Anyway, do we need a separate function?. if so, shouldn't we place it after the docstring or as a subfunction?

File: R/frequency_encoder.R [hunk 5]
```diff
@@ -21,8 +23,19 @@ encode_freq <- function(X) {
...
+      dplyr::summarise(freq = n())
+    encodings[[sym(cat)]] <- col_df
+  }
+  train_processed <- X_train
+    #dplyr::mutate_if(.tbl = X_train,
+    #                                  .predicate = ~is.element(deparse(substitute(.x)), cat_columns),
+    #                                  .fun = encode_freq, encodings)
   test_processed <- X_test
-  out <- list( "train" = train_processed, "test" = test_processed)
+  out <- list( "train" = train_processed, "test" = test_processed, "encodings" = encodings)
```
Comment (robilizando) [inline R/frequency_encoder.R]: I see what you did here, very clever

**Review by robilizando — COMMENTED** [review]
So far so good, do not forget to  include some basic defensive tests I suggested in an earlier version and 
the optional thing about the description diagram to show users how their data frame will change

**Comment by bbaillie** [comment]
Thank you Robert!

**Review by braydentang1 — APPROVED** [review]

**Pull request merged**